packages/rocketchat-federation/server/methods/index.js

```diff
@@ -0,0 +1,2 @@
+import methods from './methods';
```

Comment:
any good reason to always have an index that only imports a file with the same name as the folder? I would recommend to remove this `index.js` file, the folder `methods` and keep only the `methods.js` file (this applies to all `index.js` with same behavior :grimacing: )

Reply:
We can do this, but it introduces a new level for each require/import (`./methods/methods`), making it look strange and maybe confusing the developer, as it implies there are other files inside that same folder that could/should be imported. Also, I dislike using sole `index` files because when you have several tabs opened, you won't be able to correctly guess what are you looking at.

This is something I've learned after working on huge projects with lots of files: making the development process streamlined and well organized is a huge deal.

But I'm up to removing it if you really do not like that approach, just ping me.

Reply:
> We can do this, but it introduces a new level for each require/import (`./methods/methods`), making it look strange and maybe confusing the developer

Actually I said to remove the folder as well, so it will look the same, it will be like: `import './methods'`

> Also, I dislike using sole `index` files because when you have several tabs opened

I agree, that's why I suggested to remove the folder.. So I think if we see the need to having more files, we can create a folder which exports the same thing, so all imports will still work.

> This is something I've learned after working on huge projects with lots of files: making the development process streamlined and well organized is a huge deal.

I agree :+1: this suggestion should be in a code style guide. I'll think about this :thinking: